src/new/connection.js

```diff
@@ -0,0 +1,130 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import EventEmitter from 'events';
+import networkInstance from '../network';
+import { DEFAULT_SERVERS } from '../constants';
+import version from '../version';
+import WS from '../websocket';
+
+/**
+ * This is a Connection that may be shared by one or more wallets.
+ *
+ * It has the following states:
+ * - CLOSED: When it is disconnected from the server.
+ * - CONNECTING: When it is connecting to the server.
+ * - CONNECTED: When it is connected.
+ *
+ * You can subscribe for the following events:
+ * - update-state: Fired when the state of the Wallet changes.
+ * - websocket-msg: Fired when a new message arrive from the websocket.
+ **/
+class Connection extends EventEmitter {
+  /*
+   * network {String} 'testnet' or 'mainnet'
+   * servers {Array} List of servers for the wallet to connect to, e.g. http://localhost:8080/v1a/
+   */
+  constructor({
+    network,
+    servers = [],
+  } = {}) {
+    super();
+
+    if (!network) {
+      throw Error('You must explicitly provide the network.');
+    }
+
+    networkInstance.setNetwork(network);
+    this.network = network;
+
+    this.state = Connection.CLOSED;
+    this.serverInfo = null;
+
+    this.onConnectionChange = this.onConnectionChange.bind(this);
+    this.handleWebsocketMsg = this.handleWebsocketMsg.bind(this);
```

Comment:
Where's this method?

Reply:
This is an old code, it's already fixed.